unyt/array.py

```diff
@@ -1884,7 +1884,11 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
                     conv, offset = u1.get_conversion_factor(u0, inp1.dtype)
                     new_dtype = np.dtype("f" + str(inp1.dtype.itemsize))
                     conv = new_dtype.type(conv)
-                    if offset is not None:
+                    if (
+                        offset is not None
+                        and u1.base_offset != 0.0
+                        and not repr(u0).startswith("delta_")
```

Comment:
I'm pretty sure this last conditional will always be `True` because `delta_degF` and `delta_degC` have `base_offset=0.0`.

Reply:
the second condition in on `u1` while this one is on `u0`. I don't like it very much, I think I reached the point where I was more or less typing random conditions until I got one that passed tests, but I can guarantee that it's needed :)